cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -470,6 +485,51 @@ open class Node @Inject constructor(private val project: Project) {
         }
     }
 
+
+    private fun createSchemasCmd() = listOfNotNull(
+            Paths.get(System.getProperty("java.home"), "bin", "java").toString(),
+            "-jar",
+            "corda.jar",
+            "run-migration-scripts",
+            if (allowHibernateToManageAppSchema) "--allow-hibernate-to-manage-app-schema" else null)
+
+    private fun runSchemaMigration(){
+        if (!runSchemaMigration) return
+        project.logger.lifecycle("Run database schema migration scripts${if(allowHibernateToManageAppSchema) " - managing CorDapp schemas with hibernate" else ""}")
+        runNodeJob(createSchemasCmd(), "node-schema-cordform.log")
+    }
+
+    val LOGS_DIR_NAME: String = "logs"
+
+    private fun runNodeJob(command: List<String>, logfileName: String) {
+        val logsDir = Files.createDirectories(nodeDir.toPath().resolve(LOGS_DIR_NAME))
+        val nodeRedirectFile = logsDir.resolve(logfileName).toFile()
+        val process = ProcessBuilder(command)
+                .directory(nodeDir)
+                .redirectErrorStream(true)
+                .redirectOutput(nodeRedirectFile)
+                .apply { environment()["CAPSULE_CACHE_DIR"] = "../.cache" }
```

Comment:
may I ask what is this for?

Reply:
Hmm - that is copied and pasted 😃 
It creates the capsule cache for this run locally rather than in the user's home directory, which means it gets wiped when running e.g. `./gradlew clean`. But I am not overly attached to it.

Reply:
But it does also mean that your `corda-4.5-SNAPSHOT.jar` is the SNAPSHOT you actually asked for, and hasn't been contaminated by (say) last week's 4.5-SNAPSHOT which was lingering in your `$HOME/.capsule` directory.